### PR TITLE
Fixes #24285: Empty role in users file is logged as warning

### PR DESCRIPTION
**Pull request by clarktsiory**
https://issues.rudder.io/issues/24285

### CODE REVIEW

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/Authorizations.scala [hunk 1]
```diff
@@ -533,10 +533,11 @@ object RudderRoles {
 
   // Utility method for parsing several roles at once
   // No consistency enforcement other than if at least on "NoRight" exists, then only NoRight is returned.
-  // Unknown roles are filtered out (with log)
+  // Unknown roles are filtered out (with log), empty string are ignored.
   def parseRoles(roles: List[String]): UIO[List[Role]] = {
+    val nonEmptyRoles = roles.filter(_.nonEmpty)
```
Comment (fanf) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/Authorizations.scala:538]: I think it needs to be `.trim` before checking for non-emptyness

**Comment by clarktsiory** [comment]
PR updated with a new commit

**Review by fanf — APPROVED** [review]

**Comment by fanf** [comment]
OK, squash merging this PR

**Comment by fanf** [comment]
OK, merging this PR

**Pull request merged**